provider/provider.go

```diff
@@ -41,6 +42,7 @@ func init() {
 	// IMPORTANT: this line cannot be changed, because it's used for
 	// internal purposes at Databricks.
 	useragent.WithProduct("databricks-tf-provider", common.Version())
+	logging.SetLogger()
```

Comment:
I don't know that this is great to do in `init()`, since this assumes that the logging configuration is done fully statically by TF (and I'm a bit nervous to put things in `init()` that don't have to be there). Maybe we move this to ConfigureContextFunc.